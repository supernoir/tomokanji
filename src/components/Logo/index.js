import React from 'react';

const Logo = () => {
	return (
		<div className="brand">
			<p className="logo">
				友
				<span className="highlight">カンジ</span>
			</p>
		</div>
	);
};

export default Logo;
