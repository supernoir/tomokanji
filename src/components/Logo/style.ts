import styled from 'styled-components';

export const StyledHeader = styled.div`
  background: ${props => props.theme.black};
  opacity: 0.8;
  height: 3em;
  padding: 1em;
  display: flex;
  flex-direction:row;
  justify-content: center;
`;

export const StyledBrand = styled.span`
  margin: 0;
  color: ${props => props.theme.white};
  font-size: 2em;
  font-weight: bold;
`;

export const StyledLogo = styled.span`
  img {
  height: 6em;
  width: auto;
  }
`
//text - shadow: 0 1px 0 #ccc, 0 2px 0 #c9c9c9, 0 3px 0 #bbb, 0 4px 0 #b9b9b9, 0 5px 0 #aaa, 0 6px 1px rgba(0, 0, 0, .1), 0 0 5px rgba(0, 0, 0, .1), 0 1px 3px rgba(0, 0, 0, .3), 0 3px 5px rgba(0, 0, 0, .2), 0 5px 10px rgba(0, 0, 0, .25), 0 10px 10px rgba(0, 0, 0, .2), 0 20px 20px rgba(0, 0, 0, .15);

