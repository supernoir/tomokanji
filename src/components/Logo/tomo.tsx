import React from 'react'

const TomoSVG: React.FC = () => <img src="./../../../public/img/tomo.svg" />

export default TomoSVG