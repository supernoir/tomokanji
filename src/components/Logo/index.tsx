import React from 'react';
import { StyledHeader, StyledBrand, StyledLogo } from './style';
import TomoSVG from './tomo';

export const Header = () => {
	return (<StyledHeader>
		<StyledBrand>
			トモ
		</StyledBrand>
		<StyledLogo>
			<TomoSVG />
		</StyledLogo>
		<StyledBrand>
			カンジ
		</StyledBrand>
	</StyledHeader>)
}