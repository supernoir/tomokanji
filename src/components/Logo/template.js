import React from 'react';
import { StyledHighlight, StyledLogo } from './style';

export const Brand = ({ className }) => {
	return (
		<div className={className}>
			<StyledLogo />
		</div>
	);
};

export const Logo = ({ className }) => {
	return (
		<p className={className}>
			<span>友</span><StyledHighlight />
		</p>
	);
};

export const Highlight = ({ className }) => {
	return <span className={className}>カンジ</span>;
};
