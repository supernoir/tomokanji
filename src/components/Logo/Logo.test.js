import React from 'react';
import Logo from './index';
import { render, shallow } from 'enzyme';

describe('Logo', () => {
	test('should match the snapshot', () => {
		const component = render(<Logo />);
		expect(component).toMatchSnapshot();
	});

	test('should have a className of brand', () => {
		const component = shallow(<Logo />);
		expect(component.find('brand')).toBeTruthy();
	});
	test('should have a child with className of logo', () => {
		const component = shallow(<Logo />);
		expect(component.children().find('logo')).toBeTruthy();
	});
});
