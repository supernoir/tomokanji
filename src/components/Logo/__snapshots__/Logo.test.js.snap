// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should match the snapshot 1`] = `
<div
  class="brand"
>
  <p
    class="logo"
  >
    友
    <span
      class="highlight"
    >
      カンジ
    </span>
  </p>
</div>
`;
